DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/use_library.cpp

```diff
@@ -0,0 +1,112 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "lib_rtl.hpp"
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope.
+// This FPGA best practice reduces name mangling in the optimization report.
+class KernelComputeRTL;
+
+using MyInt27 = ac_int<27, false>;
+using MyInt54 = ac_int<54, false>;
+
+
+// Using host pipes to stream data in and out of kernal
+// IDPipeA and IDPipeB will be written at host, and then read data in kernel (device)
+// IDPipeC will be written at kernel (device) respectively, and then read data by host
+class IDPipeA;
+using InputPipeA = sycl::ext::intel::experimental::pipe<IDPipeA, unsigned>;
+class IDPipeB;
+using InputPipeB = sycl::ext::intel::experimental::pipe<IDPipeB, unsigned>;
+class IDPipeC;
+using OutputPipeC = sycl::ext::intel::experimental::pipe<IDPipeC, unsigned long>;
```

Comment:
```suggestion
using InputPipeA = sycl::ext::intel::experimental::pipe<IDPipeA, uint32_t>;
class IDPipeB;
using InputPipeB = sycl::ext::intel::experimental::pipe<IDPipeB, uint32_t>;
class IDPipeC;
using OutputPipeC = sycl::ext::intel::experimental::pipe<IDPipeC, uint64_t>;
```
use types included from stdint.h for consistent behaviour on windows